lib/ansible/modules/cloud/misc/qemu_img.py

```diff
@@ -0,0 +1,221 @@
+
+ANSIBLE_METADATA = { 'status': ['preview'],
+                     'supported_by': 'community',
+                     'version': '1.0'}
+
+DOCUMENTATION = '''
+---
+author:
+  - Jeroen Hoekx (@jhoekx)
+  - Toshaan Bharvani (@toshywoshy)
+module: qemu_img
+short_description: Create qemu images
+description:
+  - "This module creates images for qemu."
+version_added: "2.3"
+options:
+  dest:
+    description:
+    - The image file to create or remove
+    required: true
+  format:
+    description:
+    - The image format - default qcow2
+    default: qcow2
+  options:
+    description:
+    - Comma separated list of format specific options in a name=value format.
+  size:
+    description:
+    - The size of the image
+  grow:
+    choices: [ "yes", "no" ]
+    defaults: "yes"
+    description:
+    - Whether the image should grow
+  shrink:
+    choices: [ "yes", "no" ]
+    defaults: "no"
+    description:
+    - Whether the image should shrink
+  state:
+    choices: [ "absent", "present"]
+    description:
+    - If the image should be present or absent - default present
+notes:
+  - This module does not change the type/format of the image
+  - This module does not take snapshots, and should be implemented seperate
+'''
+
+EXAMPLES = '''
+# Create a raw image of 5M.
+- qemu_img:
+    dest: /tmp/testimg
+    size: 5M
+    format: raw
+
+# Enlarge the image to 6G.
+- qemu_img:
+    dest: /tmp/testimg
+    size: 6G
+    format: qcow2
+
+# Shrink the image by 3G
+    code: qemu_img
+    dest: /tmp/testing
+    size: -3G
+    shrink: yes
+    format: qcow2
+
+# Remove the image
+- qemu_img:
+    dest: /tmp/testimg
+    state: absent
+'''
+
+RETURN = '''
+# create qemu image
+present:
+  type: string
+  sample: "success"
+  returned: success
+# remove qemu image
+absent:
+  type: string
+  sample: "success"
+  returned: success
+# resize qemu image
+resize:
+  type: string
+  sample: "success"
+  returned: success
+'''
+
+import os
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.pycompat24 import get_exception
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec = dict(
+            dest = dict(type='str', required=True),
+            options = dict(type='str', default='preallocation=metadata'),
+            format = dict(type='str', default='qcow2'),
+            size = dict(type='str'),
+            grow= dict(type="bool", default=True),
+            shrink = dict(type="bool", default=False),
+            state = dict(type='str', choices=['absent', 'present'], default='present'),
+        ),
+        supports_check_mode = True,
+    )
+
+    result = dict(
+        changed = False,
+    )
+
+    state = module.params['state']
+    dest = module.params['dest']
+    img_format = module.params['format']
+    img_options = module.params['options']
+    size = module.params['size']
+    grow = module.boolean(module.params['grow'])
+    shrink = module.boolean(module.params['shrink'])
```

Comment:
Since the parameters were strictly typed to be of type "bool", we don't have to run module.boolean on the values.